psycopg/psycopg/generators.py

```diff
@@ -251,28 +253,21 @@ def _pipeline_communicate(
 
 
 def _consume_notifies(pgconn: PGconn) -> None:
-    # Consume notifies
-    while True:
-        n = pgconn.notifies()
-        if not n:
-            break
-        if pgconn.notify_handler:
-            pgconn.notify_handler(n)
+    # Consume notifies. The lock ensures that notifies
+    # are consumed and handled atomically.
+    with lock:
+        while True:
+            n = pgconn.notifies()
+            if not n:
+                break
+            if pgconn.notify_handler:
+                pgconn.notify_handler(n)
 
 
-def notifies(pgconn: PGconn) -> PQGen[List[pq.PGnotify]]:
+def notifies(pgconn: PGconn) -> PQGen[None]:
```

Comment:
Even if the main user of this function doesn't make use of this return value anymore, I don't see a reason to change this function's signature.

Reply:
There is no return value now – so _mypy_ would complain if we provided anything else.